Applications/QuickShow/QSWidget.cpp

```diff
@@ -40,6 +43,126 @@ QSWidget::~QSWidget()
 {
 }
 
+void QSWidget::clear()
+{
+    m_bitmap = nullptr;
+    m_path = {};
+
+    on_scale_change(100);
+    update();
+}
+
+void QSWidget::flip(FlipDirection flip_direction)
+{
+    auto w = m_bitmap->width();
+    auto h = m_bitmap->height();
+
+    auto new_bitmap = Gfx::Bitmap::create_purgeable(m_bitmap->format(), { w, h });
+
+    for (int i = 0; i < w; i++) {
+        for (int j = 0; j < h; j++) {
+            Color color = bitmap()->get_pixel(i, j);
+            if (flip_direction == FlipDirection::Vertical)
+                new_bitmap->set_pixel(i, h - j - 1, color);
+            else
+                new_bitmap->set_pixel(w - i - 1, j, color);
+        }
+    }
+
+    m_bitmap = new_bitmap;
+    set_scale(m_scale);
+
+    resize_window();
+}
+
+void QSWidget::rotate(RotationDirection rotation_direction)
+{
+    auto w = m_bitmap->width();
+    auto h = m_bitmap->height();
+
+    auto new_bitmap = Gfx::Bitmap::create_purgeable(m_bitmap->format(), { h, w });
```

Comment:
Since we're not using the purgeable feature, let's just use `Gfx::Bitmap::create()`